satpy/tests/reader_tests/test_abi_l1b.py

```diff
@@ -269,6 +269,42 @@ def test_vis_calibrate(self):
                          'Bidirectional Reflectance')
 
 
+class Test_NC_ABI_L1B_raw_cal(Test_NC_ABI_L1B_Base):
+    """Test the NC_ABI_L1B reader raw calibration."""
+
+    def setUp(self):
+        """Create fake data for the tests."""
+        rad_data = (np.arange(10.).reshape((2, 5)) + 1.)
+        rad_data = (rad_data + 1.) / 0.5
+        rad_data = rad_data.astype(np.int16)
+        rad = xr.DataArray(
+            rad_data,
+            dims=('y', 'x'),
+            attrs={
+                'scale_factor': 0.5,
+                'add_offset': -1.,
+                '_FillValue': 20,
+            }
+        )
+        super(Test_NC_ABI_L1B_raw_cal, self).setUp(rad=rad)
+
+    def test_raw_calibrate(self):
+        """Test RAW calibration."""
+        res = self.reader.get_dataset(
+            make_dataid(name='C05', calibration='counts'), {})
+
+        # We expect the raw data to be unchanged
+        expected = res.data
+        self.assertTrue(np.allclose(res.data, expected, equal_nan=True))
```

Comment:
Could you add an assertion here that `res.data.dtype == np.int16`? Basically to make sure the data isn't cast to another data type. I think `==` is the preferred way of comparing dtypes now. Or maybe it should be `res.data.dtype.type`.